pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -521,15 +519,15 @@ def apply_lightning(self, operations):
             """
             # Skip over identity operations instead of performing
             # matrix multiplication with the identity.
-            skipped_ops = ["Identity"]
-            invert_param = False
             for ops in operations:
-                if str(ops.name) in skipped_ops:
-                    continue
-                name = ops.name
                 if isinstance(ops, Adjoint):
                     name = ops.base.name
                     invert_param = True
+                else:
+                    name = ops.name
+                    invert_param = False
+                if name == "Identity":
+                    continue
```

Comment:
I think we should check if `isinstance(qml.Identity)` for performance, but let's do in another PR.

Reply:
That's a good call!

Reply:
yeah, I have a WIP branch replacing these strings with `isinstances` (started on Friday, but didn't get the chance to finish) as I noticed in several places we compare name strings in Python APIs.